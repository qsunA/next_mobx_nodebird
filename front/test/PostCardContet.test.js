import React from 'react';
import {mount} from 'enzyme';
import PostCardContent from '../components/PostCardContent';

const postContent = 'hello #test';

describe('<PostCardContent/>',()=>{
    it('matches snapshot',()=>{
        const wrapper = mount(<PostCardContent postData={postContent}/>);
        expect(wrapper).toMatchSnapshot();
    });
    it('renders postContent' ,()=>{
        const wrapper = mount(<PostCardContent postData={postContent}/>);
        expect(wrapper.props().postData).toBe('hello #test');
        const link = wrapper.find('Link');
        expect(link.contains('#test')).toBe(true);
    });
});