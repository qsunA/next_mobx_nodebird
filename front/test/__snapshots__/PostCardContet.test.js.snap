// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<PostCardContent/> matches snapshot 1`] = `
<PostCardContent
  postData="hello #test"
>
  <div>
    hello 
    <Link
      as="/hashtag/test"
      href={
        Object {
          "pathname": "/hashtag",
          "query": Object {
            "tag": "test",
          },
        }
      }
      key="#test"
      prefetch={true}
    >
      <a
        href="/hashtag/test"
        onClick={[Function]}
        onMouseEnter={[Function]}
      >
        #test
      </a>
    </Link>
  </div>
</PostCardContent>
`;
